### DISCUSSION TITLE
Correlation and parameterization in Fortio

### DISCUSSION THREAD

**Discussion by romanvalenciao** [fortio/fortio#554] [Q&A] [answered]
Hi guys, I reviewed the documentation and some examples but couldn't find a good example on correlation and csv parameterization with Fortio. 

- Do we have a way to perform one call, capturing dynamic values and passing to next calls (like session id, token auth, etc)?
- Do we have a way to create csv data pools to use during load test to use sequential, unique or shared hundreds or thousands of different values (like user ids, account ids, etc.)?

Thanks,
Roman V.

**Comment by ldemailly** [comment, marked as answer]
Good questions!


> Do we have a way to perform one call, capturing dynamic values and passing to next calls (like session id, token auth, etc)?

So out of the box when using just the command line, you can't chain calls but that's easy to do with fortio as a library - or even by collecting whatever you need from 1 first call and passing it as Header etc to the actual load call

We could also add a cookie jar to the standard client if you think that'd be valuable (for cookies changing over time during a single request sequence)

> Do we have a way to create csv data pools to use during load test to use sequential, unique or shared hundreds or thousands of different values (like user ids, account ids, etc.)?

The only dynamic request support from the client currently is using `{{uuid}}` in the query args or payload which generates a uuid replacement (but you don't today get to pick what that looks like; just where it is placed in your payload)

To get a handful of profiles you would use fortio as a library or multiple command line invocation; but that wouldn't work if you really are trying to reply a unique sequence of all different but specific requests

**Reply by romanvalenciao** [reply]
Thank you for the response, I'll look into fortio as a library.
Regards,
